ansible_events_ui/api/__init__.py

```diff
@@ -102,6 +102,62 @@ async def websocket_endpoint2(
                 )
                 await db.execute(query)
                 await db.commit()
+            elif data_type == "Action":
+                activation_id = data.get("activation_id")
+                if activation_id:
+                    query = select(models.activation_instances).where(
+                        models.activation_instances.c.id == int(activation_id)
```

Comment:
Why `int(...)` ad-hoc type coercion is used here?

1. First of all better to convert it at the same time you're receiving the value (e.g. at line 106) and not where the value is used.
2. It would be way better if you could utilize `pydantic` library for data validation and de-serialization.
    - It simplifies the code and allows performing data validation at the time the data is loaded and not where the data is used.
    - It can validate not only individual fields, but also a message as a whole.